eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/BeaconServiceV2.java

```diff
@@ -0,0 +1,356 @@
+import uk.ac.ebi.eva.commons.core.models.VariantSource;
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+import uk.ac.ebi.eva.commons.mongodb.entities.VariantMongo;
+import uk.ac.ebi.eva.commons.mongodb.filter.FilterBuilder;
+import uk.ac.ebi.eva.commons.mongodb.filter.VariantRepositoryFilter;
+import uk.ac.ebi.eva.commons.mongodb.services.VariantSourceService;
+import uk.ac.ebi.eva.commons.mongodb.services.VariantWithSamplesAndAnnotationsService;
+import uk.ac.ebi.eva.lib.eva_utils.DBAdaptorConnector;
+import uk.ac.ebi.eva.lib.eva_utils.MultiMongoDbFactory;
+
+import javax.servlet.http.HttpServletResponse;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static uk.ac.ebi.eva.commons.beacon.models.BeaconAlleleRequest.IncludeDatasetResponsesEnum;
+
+@Service
+public class BeaconServiceV2 {
+
+    private static final String ORGANIZATION_ID = "EMBL-EBI-EVA";
+
+    private static final String ORGANIZATION_NAME = "European Variation Archive (EMBL-EBI)";
+
+    private static final String ORGANIZATION_DESCRIPTION = "EMBL-EBI makes the world's public biological data freely" +
+            " available to the scientific community via a range of services and tools, performs basic research" +
+            " and provides professional training in bioinformatics. The European Variation Archive is an " +
+            "open-access" + " database of all types of genetic variation data from all species.";
+
+    private static final String ORGANIZATION_ADDRESS = "Wellcome Genome Campus, Hinxton, Cambridgeshire, CB10 1SD, " +
+            "United Kingdom";
+
+    private static final String ORGANIZATION_WELCOME_URL = "www.ebi.ac.uk/eva";
+
+    private static final String ORGANIZATION_CONTACT_URL = "contactUrlString";
+
+    private static final String ORGANIZATION_LOGO_URL = "www.ebi.ac.uk/eva/img/eva_logo.png";
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Autowired
+    private VariantSourceService variantSourceService;
+
+    public BeaconServiceV2() {
+    }
+
+    public BeaconImpl getBeacon() {
+        BeaconImpl beacon = new BeaconImpl();
+        beacon.setDatasets(getAllBeaconDatasets());
+        return beacon;
+    }
+
+    private List<BeaconDataset> getAllBeaconDatasets() {
+        List<BeaconDataset> beaconDatasets = new ArrayList<>();
+        beaconDatasets.addAll(getBeaconDatasetsPerDatabase("hsapiens_grch37", "randomDescription", "GRCh37",
+                "updatedatetime", "externalurl"));
+        beaconDatasets.addAll(getBeaconDatasetsPerDatabase("hsapiens_grch38", "randomDescription", "GRCh38",
+                "updatedatetime", "externalurl"));
+        return beaconDatasets;
+    }
+
+    private List<BeaconDataset> getBeaconDatasetsPerDatabase(String db, String description, String assemblyId,
```

Comment:
Since the database name and the assembly partially overlap, instead of providing `db` and `assemblyId`, this method could receive `species` and `assembly`. The database name would be composed from those 2.

Reply:
Fixed :+1: